keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -450,157 +487,135 @@ about permitting exports from `kube-system` and `default`. As a best practice,
+  // +optional
+  Status ServiceImportStatus `json:"status,omitempty"`
 }
 
-// ServiceImportSpec contains the current status of an imported service and the
-// information necessary to consume it
+// ServiceImportType designates the type of a ServiceImport
+type ServiceImportType string
+
+const (
+  // SuperclusterIP are only accessible via the Supercluster IP.
+  SuperclusterIP ServiceImportType = "SuperclusterIP"
+  // Headless services allow backend pods to be addressed directly.
+  Headless ServiceImportType = "Headless"
+)
+
+// ServiceImportSpec describes an imported service and the information necessary to consume it.
 type ServiceImportSpec struct {
-  // +patchStrategy=merge
-  // +patchMergeKey=port
-  // +listType=map
-  // +listMapKey=port
-  // +listMapKey=protocol
+  // +listType=atomic
   Ports []ServicePort `json:"ports"`
+  // +optional
+  IP string `json:"ip,omitempty"`
+  // +optional
+  Type ServiceImportType `json:"type"`
+  // +optional
+  SessionAffinity corev1.ServiceAffinity `json:"sessionAffinity"`
+  // +optional
+  SessionAffinityConfig *corev1.SessionAffinityConfig `json:"sessionAffinityConfig"`
+}
+
+// ServicePort represents the port on which the service is exposed
+type ServicePort struct {
+  // The name of this port within the service. This must be a DNS_LABEL.
+  // All ports within a ServiceSpec must have unique names. When considering
+  // the endpoints for a Service, this must match the 'name' field in the
+  // EndpointPort.
+  // Optional if only one ServicePort is defined on this service.
+  // +optional
+  Name string `json:"name,omitempty"`
+
+  // The IP protocol for this port. Supports "TCP", "UDP", and "SCTP".
+  // Default is TCP.
+  // +optional
+  Protocol Protocol `json:"protocol,omitempty"`
+
+  // The application protocol for this port.
+  // This field follows standard Kubernetes label syntax.
+  // Un-prefixed names are reserved for IANA standard service names (as per
+  // RFC-6335 and http://www.iana.org/assignments/service-names).
+  // Non-standard protocols should use prefixed names such as
+  // mycompany.com/my-custom-protocol.
+  // Field can be enabled with ServiceAppProtocol feature gate.
+  // +optional
+  AppProtocol *string `json:"appProtocol,omitempty"`
+
+  // The port that will be exposed by this service.
+  Port int32 `json:"port"`
+}
+
+// ServiceImportStatus describes derived state of an imported service.
+type ServiceImportStatus struct {
   // +optional
   // +patchStrategy=merge
   // +patchMergeKey=cluster
   // +listType=map
   // +listMapKey=cluster
- Clusters []ClusterSpec `json:"clusters"`
- // +optional
- IPFamily corev1.IPFamily `json:"ipFamily"`
- // +optional
- IP string `json:"ip,omitempty"`
+  Clusters []ClusterStatus `json:"clusters"`
 }
 
-// ClusterSpec contains service configuration mapped to a specific cluster
-type ClusterSpec struct {
+// ClusterStatus contains service configuration mapped to a specific source cluster
+type ClusterStatus struct {
```

Comment:
Could you clarify what is meant by "service configuarion" here? From example it just looks like list of clusters that have exported this service. Any specific reason to move this from Spec into Status?